From 16d637796bbab713640a2c511991772107192ece Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 20 Jan 2022 09:39:55 +0200
Subject: [PATCH 16/64] tools: s32gen1image: Add MBR field for SD boot

MBR region should be reserved for SD or eMMC boot.

Issue: ALB-8303
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 tools/s32gen1image.c | 7 +++++++
 tools/s32gen1image.h | 1 +
 2 files changed, 8 insertions(+)

diff --git a/tools/s32gen1image.c b/tools/s32gen1image.c
index 697de8264a..09fccc741b 100644
--- a/tools/s32gen1image.c
+++ b/tools/s32gen1image.c
@@ -13,6 +13,9 @@
 
 #define UNSPECIFIED	-1
 
+#define MBR_OFFSET			0x0
+#define MBR_SIZE			0x200
+
 #define S32GEN1_QSPI_IVT_OFFSET		0x0
 #define S32GEN1_SD_IVT_OFFSET		0x1000
 #define IVT_VERSION			(0x60)
@@ -133,6 +136,10 @@ static const char * const dcd_cmds[] = {
 static struct image_config iconfig;
 
 static struct program_image image_layout = {
+	.mbr_reserved = {
+		.offset = MBR_OFFSET,
+		.size = MBR_SIZE,
+	},
 	.ivt = {
 		.size = sizeof(struct ivt),
 	},
diff --git a/tools/s32gen1image.h b/tools/s32gen1image.h
index d2cb741313..974aebda3f 100644
--- a/tools/s32gen1image.h
+++ b/tools/s32gen1image.h
@@ -106,6 +106,7 @@ struct image_comp {
 };
 
 struct program_image {
+	struct image_comp mbr_reserved;
 	struct image_comp ivt;
 	struct image_comp qspi_params;
 	struct image_comp dcd;
-- 
2.17.1

